File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {

    public static final int TOTAL_OVEN_MINUTES = 40;
    public static final int MINUTES_PER_LAYER = 2;

    public int expectedMinutesInOven() {
        return TOTAL_OVEN_MINUTES;
    }

    public int remainingMinutesInOven(int actualMinutesInOven) {
        return expectedMinutesInOven() - actualMinutesInOven;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * MINUTES_PER_LAYER;
    }

    public int totalTimeInMinutes(int numberOfLayers, int actualMinutesInOven) {
        return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
    }
}

File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven(), do: 40

  def remaining_minutes_in_oven(actual_minutes) do
    expected_minutes_in_oven() - actual_minutes
  end

  def preparation_time_in_minutes(number_of_layers) do
    minutes_per_layer = 2;
    minutes_per_layer * number_of_layers
  end

  def total_time_in_minutes(number_of_layers, minutes_in_oven) do
    preparation_time_in_minutes(number_of_layers) + minutes_in_oven
  end

  def alarm() do
    "Ding!"
  end
end
